nautobot/core/tests/test_graphql.py

```diff
@@ -1444,7 +1456,14 @@ def test_query_frontport_filter_third_level(self):
                 query = "query { sites{ devices{ frontports(" + filterv + "){ id }}}}"
                 result = self.execute_query(query)
                 self.assertIsNone(result.errors)
-                self.assertEqual(len(result.data["sites"][0]["devices"][0]["frontports"]), nbr_expected_results)
+                non_empty_index = 0
+                while non_empty_index < len(result.data["sites"]):
+                    if result.data["sites"][non_empty_index]["devices"]:
+                        break
+                    non_empty_index += 1
+                self.assertEqual(
+                    len(result.data["sites"][non_empty_index]["devices"][0]["frontports"]), nbr_expected_results
+                )
```

Comment:
This is difficult for me to parse. What do you think of using enumerate?

```suggestion
                for count, _ in enumerate(result.data["sites"]):
                    if result.data["sites"][count]["devices"]:
                        self.assertEqual(
                            len(result.data["sites"][count]["devices"][0]["interfaces"]), nbr_expected_results
                        )
                        break
```